WordPress/UITestsFoundation/Screens/MySitesScreen.swift

```diff
@@ -41,4 +45,12 @@ public class MySitesScreen: ScreenObject {
         app.cells[title].tap()
         return try MySiteScreen()
     }
+
+    public func closeModalIfNeeded() {
+        if addSelfHostedSiteButtonGetter(app).isHittable && XCUIDevice.isPhone {
+            app.buttons.matching(identifier: "Cancel").element(boundBy: 1).tap()
+        }
+        if addSelfHostedSiteButtonGetter(app).isHittable && XCUIDevice.isPad { navigateBack() }
+        if cancelButtonGetter(app).isHittable { cancelButtonGetter(app).tap() }
+    }
```

Comment:
A few notes from my side for this part:

1. We're either on iPhone or on iPad, so the first two `if` blocks as mutually exclusive.
2. On iPad, `navigateBack()` swipes in this case. I think it's possible to have the code just tap the centre of `MySitesScreen`, and it ~will~ should work for both iPad and iPhone 🤔

Reply:
Thanks for reviewing it, @pachlava! 

> 1. We're either on iPhone or on iPad, so the first two if blocks as mutually exclusive.

Yeah, it was me trying to avoid 'hadouken' code with nested ifs but your second point has the key.

> 2. On iPad, navigateBack() swipes in this case. I think it's possible to have the code just tap the centre of MySitesScreen, and it will should work for both iPad and iPhone

Excellent point. I've updated the code to be device agnostic. [Test PR confirmed it works as expected](https://github.com/wordpress-mobile/WordPress-iOS/pull/17937#issuecomment-1036424498).